p2p/p2p/src/connection_maintainer.rs

```diff
@@ -104,12 +104,12 @@ where
         clippy::significant_drop_tightening
     )]
     async fn connect_to_random_seeds(&mut self) -> Result<(), OutboundConnectorError> {
-        let seeds = N::SEEDS.choose_multiple(&mut thread_rng(), MAX_SEED_CONNECTIONS);
+        let seeds = self
+            .config
+            .seeds
+            .choose_multiple(&mut thread_rng(), MAX_SEED_CONNECTIONS);
 
-        assert!(
-            seeds.len() != 0,
-            "No seed nodes available to get peers from"
-        );
+        assert_ne!(seeds.len(), 0, "No seed nodes available to get peers from");
```

Comment:
nit: https://rust-lang.github.io/rust-clippy/master/index.html#/len_zero

```suggestion
        assert!(!seeds.is_empty(), "No seed nodes available to get peers from");
```

Reply:
`seeds` is a type which doesn't have a `is_empty` stable: https://doc.rust-lang.org/nightly/core/iter/traits/exact_size/trait.ExactSizeIterator.html#method.is_empty